src/content/post/20170213_nodemcu_espruino_step_by_step.md

```diff
@@ -0,0 +1,134 @@
+title       = "Espruino on NodeMCU (step-by-step guide)"
+tags        = ['iot', 'javascript', 'espruino', 'nodemcu']
+topics      = ['IoT']
+description = "How to flash NodeMCU with Espruino firmware and basic examples on how to use it."
+slug        = espruino-nodemcu-step-by-step'
+author      = "Marco"
++++
+[SparkFabrik]: http://www.sparkfabrik.com/  "SparkFabrik"
+
+## Introduction
+This step-by-step guide is going to help you setting up and use Espruino on NodeMCU device.
+[NodeMCU](http://www.nodemcu.com/index_en.html) is:
+> An open-source firmware and development kit that helps you to prototype your IOT product within a few Lua script lines.
+We chosen the NodeMCU because it uses the ESP8266 chip which is one of the most cheap wifi module.
+
+## Buy all the things!
+Everything for this tutorial has been bought on Amazon (fast shipment reason :) )
+If you're not in a hurry, you could have a look on Aliexpress.
+
+[NodeMCU](https://www.amazon.it/gp/product/B01GCK3J40/ref=oh_aui_detailpage_o01_s00?ie=UTF8&psc=1)
+[DHT11 sensor](https://www.amazon.it/gp/product/B0154JQRPI/ref=oh_aui_detailpage_o01_s00?ie=UTF8&psc=1)
+
+## Prepare the ground
+First of all we have to prepare everything that is needed to perform the installation and usage of the espruino.
+We would need a tool to copy the firmware into the device, we would need the firmware itself and a IDE that would facilitate us writing and uploading custom code.
+
+#### Utility
+The [esptool.py](https://github.com/themadinventor/esptool) is needed to flash the firmware into the device. Just follow the instruction on the github page.
+
+Given that python is already installed on your system, just type
+```
+pip install esptool
+```
+Once installed, the executable is available on path (linux)```/usr/local/bin/esptool.py```
+
+
+For the Board described in this tutorial, CP2102 USB to UART driver is required. This driver is already part of the Linux core and maintained since version 3.x.
+The driver for Mac OSX is available [here](https://www.silabs.com/Support%20Documents/Software/Mac_OSX_VCP_Driver.zip).
+
+
+#### Firmware
+Espruino firmware for ESP8266 can either be built from source-code that can be found on [github](https://github.com/tve/Espruino) 
+or simply install an already compiled one.
+Updates on this firmware are tracked in a forum thread http://forum.espruino.com/conversations/279176/
+
+For those not interested in discussions, just [download the latest firmware version](https://www.espruino.com/Download)
+Make sure to use the firmware compatible with ESP8266.
+
+### Flash the espruino firmware
+To set NodeMCU to flash mode, keep pressed the "Flash" button while plugging in the device into the USB.
+
+Before we actually upload the new firmware, we have to make sure everything is clean. For that, we first erase the flash on the device by executing: 
+```
+esptool.py --port /dev/ttyUSB0 erase_flash
+```
+
+After that we can upload the Espruino firmware with the following command:
```

Comment:
A link to the official documentation can do in this case. Actually telling the reader that the command is bound to change over time and heading to the the docs can add more value than a command template. Thoughts?